src/data.js

```diff
@@ -263,45 +263,154 @@ let Artifact = Entity.configure({
     name:           Entity.types.String,
     storageType:    Entity.types.String,
     contentType:    Entity.types.String,
+    details:        Entity.types.JSON,
+    expires:        Entity.types.Date,
+  },
+  context: [
+    'blobStore',
+    'privateBucket',
+    'publicBucket',
+    'monitor',
+  ],
+});
+
+/** Entity for tracking artifacts.
+ * Version 2 adds 'present' field */
+let Artifact = Entity.configure({
+  version: 2,
+  properties: {
+    taskId: Entity.types.SlugId,
+    runId: Entity.types.Number,
+    name: Entity.types.String,
+    storageType: Entity.types.String,
+    contentType: Entity.types.String,
     /**
      * Location details storageType,
      *
-     * storageType: s3
-     *   bucket:        S3 bucket that contains the object
-     *   prefix:        Prefix (path) for the object within the bucket
-     *
-     * storageType: azure
-     *   container:     Azure container that holds the blob
-     *   path:          Path to blob within container
+     * storageType: blob
+     *   provider: which provider stores this artifact (e.g. s3)
+     *   region: which region this artifact is in (e.g. us-west-2)
+     *   bucket: name of the 'folder' the artifact is stored in (e.g.
+     *     taskcluster-public-artifacts)
+     *   key: name of the 'file' in the storage (e.g. $taskid/$runid/$name)
+     *   length: As per the HTTP Content-Length header
+     *   md5: hex encoded MD5 digest of content
+     *   sha256: hex encoded SHA256 digest of content
```

Comment:
Couldn't we allign this docs as in the table format below?
Also intend text under NOTE: